### ISSUE TITLE
Qwen3-VL emb微调，训练到一半CPU OOM

### ISSUE THREAD

**Issue by zhzxlcc** [modelscope/ms-swift#8372]
### Checklist / 检查清单

- [x] I have searched existing issues, and this is a new bug report. / 我已经搜索过现有的 issues，确认这是一个新的 bug report。

### Bug Description / Bug 描述

<img width="1403" height="413" alt="Image" src="https://github.com/user-attachments/assets/0eaf4b9a-5abe-4103-8d7b-5ce146d05006" />

<img width="1490" height="316" alt="Image" src="https://github.com/user-attachments/assets/65b0e363-ccf0-4eba-9cf0-7c7402e8fcbe" />

- 32卡分布式训练，使用ZERO2。

- 训练log的最后一行loss日志：{'loss': 1.89589844, 'grad_norm': 0.04135821, 'learning_rate': 4.134e-05, 'epoch': 0.31, 'global_step/max_steps': '60/193', 'percentage': '31.09%', 'elapsed_time': '3h 36m 48s', 'remaining_time': '8h 0m 35s', 'memory(GiB)': 20.61, 'train_speed(iter/s)': 0.004612}

Rank(Worker): 16 caught unknown unhandled exception:
Exception type: OOM
Detail:
return code is -9, maybe OUT OF MEMORY, please check kmon

### How to Reproduce / 如何复现

swift_version：4.0.0.dev0
"model": "Qwen/Qwen3-VL-Embedding-8B",
    "dataset": 视频训练数据.json,
    "task_type": "embedding",
    "tuner_type": "lora",
    "lora_rank": 8,
    "lora_alpha": 32,
    "target_modules": "all-linear",
    "learning_rate": "5e-5",
    "attn_impl": "flash_attn",
    "max_length": 4096,
    "load_from_cache_file": true,
    "num_train_epochs": 1,
    "per_device_train_batch_size": 16,
    "per_device_eval_batch_size": 2,
    "gradient_accumulation_steps": 2,
    "dataloader_num_workers": 0,
    "dataset_num_proc": 1,
    "warmup_ratio": 0.05,
    "loss_type": "infonce",
    "infonce_temperature": 0.02,
    "deepspeed": "zero2",
    "padding_free": true,
    "torch_dtype": "bfloat16",
    "save_steps": 20,
    "eval_steps": 500,
    "save_total_limit": 2,
    "logging_steps": 20,
    "model_kwargs": {"fps_max_frames":6, "image_max_token_num":60, "video_max_token_num":60,"video_min_token_num":8}

**Comment by zhzxlcc** [comment]
尝试使用把视频数据export，然后使用cached数据集进行训练，在export时也报错CPU OOM？是否与环境配置或者数据量有关？两种token设置均报错。
```shell
WORLD_SIZE=32  # 数据导出任务，不需要太多 worker？8
model=Qwen/Qwen3-VL-Embedding-8B
task_type=embedding
dataset=视频数据.jsonl
output_dir=./20w_cached/
to_cached_dataset=true
load_from_cache_file=true
split_dataset_ratio=0.0
dataset_num_proc=1
max_length=4096
# model_kwargs_val='{"fps_max_frames":6, "image_max_token_num":196, "video_max_token_num":196, "video_min_token_num":32}'
model_kwargs_val='{"fps_max_frames":6, "image_max_token_num":144, "video_max_token_num":144, "video_min_token_num":16}'
```

报错如下：

<img width="705" height="186" alt="Image" src="https://github.com/user-attachments/assets/76f0c350-09e5-42e9-b154-d60fa7f91478" />

<img width="1018" height="327" alt="Image" src="https://github.com/user-attachments/assets/c23e5618-0d4b-46c1-a05b-aac2fdbbcbfa" />

<img width="1242" height="342" alt="Image" src="https://github.com/user-attachments/assets/86087710-a7cf-43d6-805c-16a8fdbc85e7" />
